dust-lambda-extension/extensions/dust

```diff
@@ -0,0 +1,294 @@
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	if [[ $http_code -lt 200 || $http_code -ge 300 ]]; then
+		LOG_ERROR "HTTP request failed with status $http_code for $method $url"
+		LOG_DEBUG "Response: $(cat "$temp_file")"
+		rm -f "$temp_file"
+		return 1
+	fi
+
+	cat "$temp_file"
+	rm -f "$temp_file"
+	return 0
+}
+
+# Handle signals cleanly with proper cleanup
+handle_signal() {
+	LOG_INFO "Received termination signal. Preparing to shut down."
+	EXIT_REQUESTED=true
+	do_shutdown
+}
+
+do_shutdown() {
+	LOG_INFO "Starting clean shutdown process."
+
+	if [[ -n $EXTENSION_ID && $EXIT_REQUESTED == "true" ]]; then
+		LOG_DEBUG "Signaling extension shutdown to Lambda."
+		# Signal the Lambda platform that we are exiting
+		if ! make_api_request "POST" "$EXIT_API" "Lambda-Extension-Identifier: $EXTENSION_ID" >/dev/null 2>&1; then
+			LOG_WARNING "Failed to notify Lambda platform of extension shutdown"
+		fi
+	fi
+
+	# Clean up temporary files
+	[[ -f $TMP_CHALK_PATH ]] && rm -f "$TMP_CHALK_PATH"
+
+	exit 0
+}
+
+# Setup binary with proper validation and error handling
+setup_chalk_binary() {
+	# Check if chalk binary exists and is executable
+	if [[ ! -x $CHALK_BIN_PATH ]]; then
+		LOG_ERROR "chalk binary not found or not executable at $CHALK_BIN_PATH"
+		return 1
+	fi
+
+	# Copy chalk binary to /tmp if it doesn't exist or is outdated
+	if [[ ! -x $TMP_CHALK_PATH ]] || [[ $CHALK_BIN_PATH -nt $TMP_CHALK_PATH ]]; then
+		LOG_INFO "Copying chalk binary to /tmp"
+		if ! cp "$CHALK_BIN_PATH" "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to copy chalk binary to /tmp"
+			return 1
+		fi
+
+		if ! chmod +x "$TMP_CHALK_PATH"; then
+			LOG_ERROR "Failed to make chalk binary executable in /tmp"
+			return 1
+		fi
+	fi
+
+	return 0
+}
+
+# Process chalk marks with better error handling
+process_chalk_marks() {
+	LOG_INFO "$TMP_CHALK_PATH extracting $CHALK_MARK_PATH"
+	if [[ -f $CHALK_MARK_PATH ]]; then
+		LOG_DEBUG "Found $CHALK_MARK_PATH, running chalk extract."
+		if ! "$TMP_CHALK_PATH" extract "$CHALK_MARK_PATH"; then
+			LOG_WARNING "chalk extract command failed"
+			return 1
+		fi
+	else
+		LOG_WARNING "No chalk.json found at $CHALK_MARK_PATH"
+		return 1
+	fi
+
+	return 0
+}
+
+# JSON parsing using pure bash/standard tools (no jq dependency)
+extract_json_field() {
+	local json="$1"
+	local field="$2"
+
+	# Use grep with improved regex to handle various JSON formatting
+	local result
+	result=$(echo "$json" | grep -o "\"$field\"[[:space:]]*:[[:space:]]*\"[^\"]*\"" |
+		sed "s/\"$field\"[[:space:]]*:[[:space:]]*\"\([^\"]*\)\"/\1/")
+
+	# Handle case where field might have null value
+	if [[ $result == "null" ]]; then
+		result=""
+	fi
+
+	echo "$result"
+}
+
+extract_extension_id() {
+	local temp_file="$1"
+	# Extract the Extension ID from response headers
+	EXTENSION_ID=$(awk '/^[Ll]ambda-[Ee]xtension-[Ii]dentifier:/ {
+        gsub(/\r$/, "");
+        gsub(/^[^:]*:[[:space:]]*/, "");
+        print $0;
+        exit
+    }' "$temp_file")
```

Comment:
in chalk setup action

https://github.com/crashappsec/setup-chalk-action/blob/44e40964dfd031b5d6b77ce5015a418c673ff649/setup.sh#L170-L177

I think awk syntax is less obvious to people generally vs simple `grep`

as `curl` header output is known, this could be easily grepped